pkg/ccl/changefeedccl/cdctest/validator_helper.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2018 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package cdctest
+
+import (
+	gojson "encoding/json"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/util/json"
+	"github.com/cockroachdb/cockroach/pkg/workload/rand"
+)
+
+func extractFieldFromJSON(d json.JSON) (interface{}, error) {
+	switch d.Type() {
+	case json.ArrayJSONType:
+		iter, _ := d.AsArray()
+		ret := make([]interface{}, 0)
+		for i := 0; i < len(iter); i++ {
+			if i, err := extractFieldFromJSON(iter[i]); err == nil {
+				ret = append(ret, i)
+			}
+		}
+		return ret, nil
+	case json.ObjectJSONType:
+		if g, err := geo.ParseGeographyFromGeoJSON([]byte(d.String())); err == nil {
+			return rand.DatumToGoSQL(tree.NewDGeography(g))
+		}
+		if g, err := geo.ParseGeometryFromGeoJSON([]byte(d.String())); err == nil {
+			return rand.DatumToGoSQL(tree.NewDGeometry(g))
+		}
+	default:
+		break
+	}
+	var ret interface{}
+	err := gojson.Unmarshal([]byte(d.String()), &ret)
+	return ret, err
+}
+
+func parseJSON(d json.JSON) (map[string]interface{}, error) {
+	if d == nil {
+		return nil, nil
+	}
+	ret := make(map[string]interface{})
+	switch d.Type() {
```

Comment:
also, the way this function is used means that the input is always an object. can you simplify it using this? i'm also not convinced using the crdb json stuff is adding value here over just using gojson. if so you could just pass a map[string]any into here and greatly simplify things

Reply:
Discussed this offline as well. Usually we take in Object, but we take in Null JSON sometimes as well. Addressed it by adding a case specifically for `case json.NullJSONType:`.

Reply:
> i'm also not convinced using the crdb json stuff is adding value here over just using gojson.

Chatted this offline as well. It is a bit tricky here since parsing it using gojson would require us to parse it to an interface first and then convert this interface back to a JSON type so that we can use geo.ParseGeographyFromGeoJSON if this interface turns out to be for a geometry type. We decided to just keep it this way for now and will revisit and see if there is a better way to handle this in the future.